data-processing-lib/doc/transform-exceptions.md

```diff
@@ -5,13 +5,20 @@ a given input data (e.g., pyarrow Table) does not have the expected format (.e.g
 In general, it should identify such conditions by raising an exception. 
 With this in mind, there are two types of exceptions:
 
-1. Those that would not allow any data to be processed (e.g. model loading problem).
-2. Those that would not allow a specific datum to be processed (e.g. missing column).
+1. Those that would not allow a specific datum to be processed (e.g. missing column).
+2. Those that would not allow any data to be processed (e.g. model loading problem).
 
-In the first situation the transform should throw an exception from the initializer, which
-will cause the runtime to terminate processing of all data. 
-In the second situation (identified in the `transform()` or `flush()` methods), the transform
-should throw an exception from the associated method.
+To fail on an individual piece of data (1 above) provided to either
+AbstractBinaryTransform's `transform_binary`, or
+AbstractTableTransform's tranform()`,
+these methods should throw any exception other than UnrecoverableException. 
 This will cause only the error-causing datum to be ignored and not written out,
-but allow continued processing of tables by the transform.
-In both cases, the runtime will log the exception as an error.
\ No newline at end of file
+but allow continued processing of data by the transform.
+
+To terminate processing of all data by the runtime (2 above), 
+the transform should either 
+* throw any exception in `__init__()`, or 
+* throw an UnrecoverableException from AbstractBinaryTransform's `transform_binary`, `flush_binary(), or
```

Comment:
throwing any exception in `__init__()` will not terminate the process. It will just continue re creating an actor

Reply:
so what happens, effectively, to the actor and the processing of data when init() throws an exception.  I would presume that no data is processed by an actor that gets an exception in init(),and since all actors likely get the same exception, all data processing is ended?

Reply:
Added more precise doc in https://github.com/IBM/data-prep-kit/pull/616